Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <SdfAssetBuilder/SdfAssetBuilder.h>
+
+#include <AzCore/IO/FileIO.h>
+#include <AzCore/IO/IOUtils.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzFramework/StringFunc/StringFunc.h>
+#include <AzToolsFramework/Entity/EntityUtilityComponent.h>
+#include <AzToolsFramework/Prefab/PrefabLoaderScriptingBus.h>
+#include <AzToolsFramework/Prefab/PrefabSystemComponentInterface.h>
+#include <AzToolsFramework/Prefab/PrefabSystemScriptingBus.h>
+#include <AzToolsFramework/Prefab/Procedural/ProceduralPrefabAsset.h>
+
+#include <AssetBuilderSDK/AssetBuilderSDK.h>
+#include <AssetBuilderSDK/SerializationDependencies.h>
+
+namespace ROS2
+{
+        namespace
+        {
+            [[maybe_unused]] constexpr const char* SdfAssetBuilderName = "SdfAssetBuilder";
+            constexpr const char* SdfAssetBuilderJobKey = "Sdf Asset Builder";
+            constexpr AZStd::array<const char*,4> SdfAssetSourceExtensions =
+            {
+                "*.sdf",
+                "*.urdf",
+                "*.world",
+                "*.xacro"
+            };
```

Comment:
The Scene API has the list of available extensions data driven via the Settings Registry.

There is a [sceneassetimporter.setreg ](https://github.com/o3de/o3de/blob/development/Registry/sceneassetimporter.setreg#L4-L14) file which specifies the valid extensions to use.
That could be useful here to follow that approach.

All that would need to be added is a call to the SettingsRegistryVistUtils VisitArray function that could just read that array for specific settings path.
Here is an example of using that api to read the "external_subdirectories" array (https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp#L1688-L1705)

Reply:
Fixed! Moved to an sdfassetbuilder_settings.setreg file and verified that it was getting applied correctly.